test/controllers/event1/PaymentNatureControllerSpec.scala

```diff
@@ -0,0 +1,137 @@
+
+  private val waypoints = EmptyWaypoints
+
+  private val formProvider = new PaymentNatureFormProvider()
+  private val form = formProvider()
+
+  private val mockUserAnswersCacheConnector = mock[UserAnswersCacheConnector]
+
+  private def getRoute: String = routes.PaymentNatureController.onPageLoad(waypoints).url
+  private def postRoute: String = routes.PaymentNatureController.onSubmit(waypoints).url
+
+  private val extraModules: Seq[GuiceableModule] = Seq[GuiceableModule](
+    bind[UserAnswersCacheConnector].toInstance(mockUserAnswersCacheConnector)
+  )
+
+  override def beforeEach: Unit = {
+    super.beforeEach
+    reset(mockUserAnswersCacheConnector)
+  }
+
+  "Test Controller" - {
+
+    "must return OK and the correct view for a GET" in {
+
+      val application = applicationBuilder(userAnswers = Some(emptyUserAnswers)).build()
+
+      running(application) {
+        val request = FakeRequest(GET, getRoute)
+
+        val result = route(application, request).value
+
+        val view = application.injector.instanceOf[PaymentNatureView]
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual view(form, waypoints)(request, messages(application)).toString
+      }
+    }
+
+    "must populate the view correctly on a GET when the question has previously been answered" in {
+
+      val userAnswers = UserAnswers().set(PaymentNaturePage, PaymentNature.values.head).success.value
+
+      val application = applicationBuilder(userAnswers = Some(userAnswers)).build()
+
+      running(application) {
+        val request = FakeRequest(GET, getRoute)
+
+        val view = application.injector.instanceOf[PaymentNatureView]
+
+        val result = route(application, request).value
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual view(form.fill(PaymentNature.values.head), waypoints)(request, messages(application)).toString
+      }
+    }
+
+    "must save the answer and redirect to the next page when valid data is submitted" in {
+      when(mockUserAnswersCacheConnector.save(any(), any(), any())(any(), any()))
+        .thenReturn(Future.successful(()))
+
+      val application =
+        applicationBuilder(userAnswers = Some(emptyUserAnswers), extraModules)
+          .build()
+
+      running(application) {
+        val request =
+          FakeRequest(POST, postRoute).withFormUrlEncodedBody(("value", PaymentNature.values.head.toString))
+
+        val result = route(application, request).value
+        val updatedAnswers = emptyUserAnswers.set(PaymentNaturePage, PaymentNature.values.head).success.value
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual PaymentNaturePage.navigate(waypoints, emptyUserAnswers, updatedAnswers).url
+      }
+    }
+
+    "must return bad request when invalid data is submitted" in {
+      when(mockUserAnswersCacheConnector.save(any(), any(), any())(any(), any()))
```

Comment:
afaik, this mock is useless here because of bad request.

and like above, you can verify that this mock is never been called in this scenario.